cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -362,20 +365,21 @@ public ApplicationDetail getLatestAppDetail(ApplicationReference appRef) throws
    * that don't exist.
    * @throws Exception if failed to get details.
    */
-  public Map<ApplicationId, ApplicationDetail> getAppDetails(Collection<ApplicationId> appIds) throws Exception {
+  public Map<ApplicationId, ApplicationDetail> getAppDetails(Collection<ApplicationId> appIds,
+                                                             boolean getSourceControlMeta) throws Exception {
```

Comment:
Don't use this call at all, it's very unsafe from memory perspective, especially for smaller instances. e.g. Developer instance may have problems with 2-3 ids

Reply:
Updated the push one single application at a time